client/components/select-vertical/index.tsx

```diff
@@ -1,41 +1,75 @@
 import { useTranslate } from 'i18n-calypso';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { useDebounce } from 'use-debounce';
 import FormLabel from 'calypso/components/forms/form-label';
-import { useSiteVerticalsQuery } from 'calypso/data/site-verticals';
+import { useSiteVerticalQuery, useSiteVerticalsQuery } from 'calypso/data/site-verticals';
 import SuggestionSearch from './suggestion-search';
 import type { Vertical } from './types';
 import type { SiteVerticalsResponse } from 'calypso/data/site-verticals';
 
 interface Props {
+	defaultVertical?: string;
 	isSkipSynonyms?: boolean;
 	onSelect?: ( vertical: Vertical ) => void;
 }
 
-const SelectVertical: React.FC< Props > = ( { isSkipSynonyms, onSelect } ) => {
+const SelectVertical: React.FC< Props > = ( { defaultVertical, isSkipSynonyms, onSelect } ) => {
 	const translate = useTranslate();
-	const [ searchTerm, setSearchTerm ] = useState( '' );
+	const [ searchTerm, setSearchTerm ] = useState< string >( '' );
+	const [ isLoading, setIsLoading ] = useState< boolean >( false );
+	const [ isDisableInput, setIsDisableInput ] = useState< boolean >( false );
 	const [ debouncedSearchTerm ] = useDebounce( searchTerm, 300 );
+	const isQueryDefaultValue = typeof defaultVertical === 'string' && defaultVertical !== '';
+	const isQuerySuggestions = debouncedSearchTerm !== '';
```

Comment:
I think that the logic to determine whether to perform the queries should be moved into the `usSiteVerticalQuery` and `useSiteVerticalsQuery` query hooks. e.g. see 
[use-block-editor-settings-query.ts:33](https://github.com/Automattic/wp-calypso/blob/trunk/client/data/block-editor/use-block-editor-settings-query.ts#L33)


To manage the loading state, the underlying useQuery hook returns an `isLoading` property e.g. 
`const { data: defaultValue, isLoading: isDefaultValueLoading } = useSiteVerticalQuery( { id: defaultVertical } );`

see [with-block-editor-settings.js:11](https://github.com/Automattic/wp-calypso/blob/trunk/client/data/block-editor/with-block-editor-settings.js#L11)

Reply:
I think it would be safer to control the enabling/disabling of queries from the component using the query hook, since the query hook itself should have querying disabled by default (see [use-block-editor-settings-query.ts](https://github.com/Automattic/wp-calypso/blob/trunk/client/data/block-editor/use-block-editor-settings-query.ts#L19)). Does it make sense?

Reply:
> ... control the enabling/disabling of queries from the component using the query hook ...

Do you mean as it is now or moving the logic? I'm saying that I think you could perform the check for default vertical within the `useSiteVerticalQuery` component. 

e.g. this line https://github.com/Automattic/wp-calypso/pull/62726/files#diff-8900c548961a42422dcf09edb638ec14278163cdb0845abc37dfa825897239fcR10
could be
`enabled: typeof defaultVertical === 'string' && defaultVertical !== '',`

I think that would have the same effect as the current code while keeping the check within the hook, which is more consistent with the other useQuery hook. I don't think that the hook needs to be disabled by default, I think that the 
useBlockEditorSettingsQuery is a bit of a special case because it's only active for logged in users.